pebblo_saferetriever/langchain/identity-rag-sharepoint/pebblo_identity_rag_sharepoint.py

```diff
@@ -0,0 +1,131 @@
+# Fill-in OPENAI_API_KEY in .env file in this directory before proceeding
+from dotenv import load_dotenv
+from sharepoint_auth import get_authorized_identities
+from langchain.chains import PebbloRetrievalQA
+from langchain.chains.pebblo_retrieval.models import AuthContext, ChainInput
+from langchain_community.document_loaders import UnstructuredFileIOLoader
+from langchain_community.document_loaders.pebblo import PebbloSafeLoader
+from langchain_community.vectorstores.qdrant import Qdrant
+from langchain_community.document_loaders.sharepoint import SharePointLoader
+from langchain_openai.embeddings import OpenAIEmbeddings
+from langchain_openai.llms import OpenAI
+import json
+import os
+
+load_dotenv()
+
+
+class PebbloIdentityRAGSharePoint:
+    def __init__(self, folder_id: str, folder_path: str, file_id: str, site_id: str, collection_name: str):
+        self.app_name = "pebblo-identity-rag-1"
+        self.folder_id = folder_id
+        self.file_id = file_id
+        self.site_id = site_id
+        self.folder_path = folder_path
+        self.collection_name = collection_name
+
+        # Load documents
+        print("\nLoading RAG documents ...")
+        self.loader = PebbloSafeLoader(
+            SharePointLoader(
+                document_library_id=self.folder_id, 
+                folder_path=self.folder_path, 
+                file_id=self.file_id, 
+                site_id=self.site_id, 
+                auth_with_token=False
+            ),
+            name=self.app_name,  # App name (Mandatory)
+            owner="Joe Smith",  # Owner (Optional)
+            description="Identity enabled SafeLoader and SafeRetrival app using Pebblo",  # Description (Optional)
+        )
+        self.documents = self.loader.load()
+        print(self.documents[-1].metadata.get("authorized_identities"))
+        print(f"Loaded {len(self.documents)} documents ...\n")
+
+        # Load documents into VectorDB
+
+        print("Hydrating Vector DB ...")
+        self.vectordb = self.embeddings()
+        print("Finished hydrating Vector DB ...\n")
+
+        # Prepare LLM
+        self.llm = OpenAI()
+        print("Initializing PebbloRetrievalQA ...")
+        self.retrieval_chain = self.init_retrieval_chain()
+
+    def init_retrieval_chain(self):
+        """
+        Initialize PebbloRetrievalQA chain
+        """
+        return PebbloRetrievalQA.from_chain_type(
```

Comment:
@arpitkumar980 Update the initialization of `PebbloSafeLoader` and `PebbloRetrievalQA` to match the [latest sample in the main branch](https://github.com/daxa-ai/pebblo/blob/main/pebblo_saferetriever/langchain/identity-n-semantic-rag/semantic_rag.py), and then we can merge this PR.